src/DDTrace/Transport/Http.php

```diff
@@ -62,19 +66,24 @@ private function configure($config)
     {
         $host = getenv(self::AGENT_HOST_ENV) ?: self::DEFAULT_AGENT_HOST;
         $port = getenv(self::TRACE_AGENT_PORT_ENV) ?: self::DEFAULT_TRACE_AGENT_PORT;
-        $path = self::DEFAULT_TRACE_AGENT_PATH;
-        $endpoint = "http://${host}:${port}${path}";
+        $defaultEndpoint = "http://${host}:${port}" . self::DEFAULT_TRACE_AGENT_PATH;
+        $prioritySamlpingEndpoint = "http://${host}:${port}" . self::PRIORITY_SAMPLING_TRACE_AGENT_PATH;
```

Comment:
I know this is probably not very bad for performance. 
But it worries me a bit that we're doing some computations every request, that are not normally used during the request and after.

Overall would be great to defer most of the transport setup to the very end when we need to actually send  (but this is probably out of the scope for this PR).

Reply:
This is a good point and I like the idea. If you agree I will do for the parts that I touched in this PR and will postpone a wider refactoring of the Http transport (as the same applies also to other config params) to a later stage.

Reply:
As discussed, it is risky as it introduces potential breaking changes for users, so I would defer it to a later http transport refactoring